### PR TITLE
Modernize Python3 code

### PR DESCRIPTION
**Pull request by eumiro**
A few independent commits, please feel free to discuss them individually:

1. access the keys of a dictionary implicitly when iterating or checking
2. get all necessary variables in the header of a loop
3. remove redundant parentheses (leave those in returning tuples though)
4. chain conditions
5. unescape characters in regexes that do not have to be escaped

### CODE REVIEW

**Comment by eumiro** [comment]
Rebased to the current `master`.

File: py3status/modules/net_iplist.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ class Py3status:
 
     def post_config_hook(self):
         self.iface_re = re.compile(r"\d+: (?P<iface>[\w\-]+):")
-        self.ip_re = re.compile(r"\s+inet (?P<ip4>[\d\.]+)(?:/| )")
+        self.ip_re = re.compile(r"\s+inet (?P<ip4>[\d.]+)(?:/| )")
```
Comment (ultrabug) [inline py3status/modules/net_iplist.py:80]: ain't that catch "any caracter" instead of the intended "dot" ?

**Review by ultrabug — COMMENTED** [review]
Thanks a lot for this amazing code QA mate

File: py3status/modules/net_iplist.py (hunk 1 above)
Comment (eumiro) [reply py3status/modules/net_iplist.py:80]: No.
https://docs.python.org/3/library/re.html#regular-expression-syntax

    Special characters lose their special meaning inside sets. For example, `[(+*)]` will match any of the literal characters `'('`, `'+'`, `'*'`, or `')'`.

Comment (eumiro) [reply py3status/modules/net_iplist.py:80]: No.
https://docs.python.org/3/library/re.html#regular-expression-syntax

    Special characters lose their special meaning inside sets. For example, `[(+*)]` will match any of the literal characters `'('`, `'+'`, `'*'`, or `')'`.

Comment (ultrabug) [reply py3status/modules/net_iplist.py:80]: wow TIL something new again <3

**Comment by ultrabug** [comment]
Thank you mate!

**Pull request merged**